contrib/wizard/src/Plugin/WizardStep/EntityFormMode.php

```diff
@@ -52,14 +67,30 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
+    /* @var WizardInterface $config */
+    $config = $this->configuration['wizard_config'];
     $entity = $this->getContextValue('entity');
+    $step = $this->configuration['step'];
 
     $form_display = $form_state->get('form_display');
     $form_display->extractFormValues($entity, $form, $form_state);
 
     $form_display->formSubmitComponents($form, $form_state);
-    $entity->save();
-    $form_display->saveFormEntities($form, $form_state);
+
+    if (!$config->shouldSaveOnFinish()) {
+      $entity->save();
+      $form_display->saveFormEntities($form, $form_state);
+    }
+    else {
+      // Only saving at the end - cache the other form entities for now.
+      $flexiform_entities = array_map(function ($wrapper) {
+        return $wrapper->getEntity();
+      }, $form_display->getFormEntityManager()->getFormEntities());
+
+      $cached_values = $form_state->getTemporaryValue('wizard');
+      $cached_values['flexiform_entities'][$step] = $flexiform_entities;
```

Comment:
@JeremySkinner what behaviour does this cause if you want to use the same entity across multiple steps?

It looks like the changes on step 1 might not be included on step 2 (or whatever step had that entity a second time). I'm trying to think of a way we could test that. Hmmm.